### PR TITLE
Timestamp real32 bug

### LINKED ISSUE #869: DateEncoderRegionTest.testLinking fails on my PC
continuing from #867 

> I'm not sure why it's (now?) failing for you, but an encoder shouldn't certainly be OS dependant.
> Do you assume it is by the comments? I just quickly went through it, imho the `// OSx ...` (and linux) comments are old/wrong. The test is normally run on all CI/platforms with the same results.
> 
> Btw, do your cpp,py tests for `datetime encoder` (not region) pass?

When i ran `python setup.py test`
it executes cpp tests, but don't get into py tests,  (that is probably expected?)

```
[==========] 386 tests from 44 test cases ran. (35713 ms total)
[  PASSED  ] 385 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] DateEncoderRegionTest.testLinking

 1 FAILED TEST
  YOU HAVE 2 DISABLED TESTS

Traceback (most recent call last):
  File "setup.py", line 448, in <module>
    "htm-bindings-check = htm.bindings.check:checkMain",
  File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3/dist-packages/setuptools/command/test.py", line 229, in run
    self.run_tests()
  File "setup.py", line 172, in run_tests
    subprocess.check_call([cpp_tests])
  File "/usr/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/home/osboxes/HTM/htm.core/build/Release/bin/unit_tests']' returned non-zero exit status 1.
```

When i manually run date_test.py in py/tests/encoders it passes.

### PR DESCRIPTION
**Pull request by Zbysekz**
Closes #869

Replaced Real32 and Real (this is global defined variable, can be Real32/Real64)  to Real64.

Now the datetimeEncoder test passes even in New York timezone (UTC-4).

### CODE REVIEW

**Review by breznak — APPROVED** [review]
thank you, that was a nasty bug! 
I wonder if there are performance implications of switching to real64, but since this fixes a bug, there's no other way. :+1:

**Comment by dkeeney** [comment]
@breznak  For a 64bit machine, the double will fit in a machine word so it should not affect speed very much.

@Zbysekz  we will let you click the Merge button.  :)

**Comment by Zbysekz** [comment]
> @Zbysekz we will let you click the Merge button. :)

I want to click that nice button, but it seems that i still don't have the permission :) (no write access)

File: src/test/unit/engine/HelloRegionTest.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ TEST(HelloRegionTest, demo) {
   std::string path = Path::join("TestOutputDir", "Data.csv");
 
   // enter some arbitrary test data.
-  std::vector<std::vector<Real32>> testdata = {{0, 1.5f, 2.5f},
+  std::vector<std::vector<Real64>> testdata = {{0, 1.5f, 2.5f},
```
[the remark this answers is not in the document]
Comment (Zbysekz) [reply src/test/unit/engine/HelloRegionTest.cpp:44]: ah ok, right, good point, f are for floats only. but as you suggest, let compiler make it

**Comment by breznak** [comment]
@Zbysekz you should now be merging like a pro! but don't go berserk on the PRs again :smiley:

**Comment by Zbysekz** [comment]
> @Zbysekz you should now be merging like a pro! but don't go berserk on the PRs again 😃

Thx, i will try ;) :D

**Pull request merged**